cli/upload/upload.go

```diff
@@ -0,0 +1,149 @@
+package upload
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/buildbuddy-io/buildbuddy/cli/arg"
+	"github.com/buildbuddy-io/buildbuddy/cli/log"
+	"github.com/buildbuddy-io/buildbuddy/cli/storage"
+	"github.com/buildbuddy-io/buildbuddy/server/remote_cache/cachetools"
+	"github.com/buildbuddy-io/buildbuddy/server/remote_cache/digest"
+	"github.com/buildbuddy-io/buildbuddy/server/util/grpc_client"
+	"google.golang.org/grpc/metadata"
+
+	repb "github.com/buildbuddy-io/buildbuddy/proto/remote_execution"
+	rspb "github.com/buildbuddy-io/buildbuddy/proto/resource"
+	bspb "google.golang.org/genproto/googleapis/bytestream"
+)
+
+var (
+	flags = flag.NewFlagSet("upload", flag.ContinueOnError)
+
+	target             = flags.String("target", "grpcs://remote.buildbuddy.io", "Cache gRPC target")
+	remoteInstanceName = flags.String("remote_instance_name", "", "Remote instance name")
+	compress           = flags.Bool("compress", true, "If true, enable compression of uploads to remote caches")
+	stdin              = flags.Bool("stdin", false, "If true, read from stdin")
+	digestFunction     = flags.String("digest_function", "SHA256", "If set, use this digest function for uploads.")
```

Comment:
Can we specify what the supported digest functions here?

Reply:
End users should not set this to anything besides SHA256 for now, so I don't think we should document these yet.